internal/injector/aspect/context/node.go

```diff
@@ -19,16 +21,32 @@ type NodeChain struct {
 	index  int
 }
 
+var pool = sync.Pool{New: func() any { return &NodeChain{} }}
+
+// Child creates a new [NodeChain] that represents the node pointed to by the provided
+// [dstutil.Cursor]. The cursor's parent must match the receiver. Values returned by
+// [NodeChain.Child] should be returned to the pool by calling [NodeChain.Release] on them, as this
+// allows for a significant reduction of allocations.
 func (n *NodeChain) Child(cursor *dstutil.Cursor) *NodeChain {
 	if n != nil && n.node != cursor.Parent() {
 		panic("cursor does not point to a child of this node")
 	}
-	return &NodeChain{
-		parent: n,
-		node:   cursor.Node(),
-		name:   cursor.Name(),
-		index:  cursor.Index(),
-	}
+
+	c, _ := pool.Get().(*NodeChain)
```

Comment:
```suggestion
	c := pool.Get().(*NodeChain)
```
It's faster without the type check right ?